src/Stratis.Bitcoin.Features.SmartContracts/SmartContractPosRuleRegistration.cs

```diff
@@ -23,6 +23,7 @@ public void RegisterRules(IConsensus consensus)
             consensus.IntegrityValidationRules = new List<IIntegrityValidationConsensusRule>()
             {
                 new BlockMerkleRootRule(),
+                new PosBlockSignatureRepresentationRule(),
```

Comment:
Yeah seems right, this is more of an integrity check.

Reply:
>move the rule in front of PosBlockSignatureRule (this by itself makes sure block is not banned, just peer, which is to avoid netsplit)

Following this comment its very correct integrity failure will ban the node not the block

Reply:
Yes, agreed